src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,158 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node head;
+    private Node tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        link(tail, value, null);
```

Comment:
It's really difficult to read the code of add methods, because the developer should always look below link method and remember what you provide in the method. Also, the logic of this two add methods are different, so please try to make changes.

For example, this add method:
```
Node<T> node = new Node<>(null, value, null);
        if (head == null) {
            head = tail = node;
        } else {
            tail.next = node;
            node.prev = tail;
            tail = node;
        }
        size++;
```